### PR TITLE
Feature/17408 version editable

### PR DESCRIPTION
**Pull request by 0xF013**
[`* `#17408` version editable`](http://community.openproject.org/work_packages/17408)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://0xf013-feature-17408-version-editable.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://0xf013-feature-17408-version-editable.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by 0xF013** [comment]
@myabc @ulferts ready for review

@ulferts please check the empty option case and if the version behaves according to whatever was in that earlier PR

File: frontend/npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,34 @@
+0 info it worked if it ends with ok
```
Comment (myabc) [inline frontend/npm-debug.log]: @0xF013 this shouldn't be in the commit!

File: frontend/tests/unit/tests/work_packages/controllers/details-tab-overview-controller-test.js [hunk 2]
```diff
@@ -420,103 +420,6 @@ describe('DetailsTabOverviewController', function() {
...
-            buildController();
-          });
-
-          it('should have an empty value', function() {
-            expect(fetchEmptyPropertiesWithName('versionName')[0].value).not.to.be.ok;
-          });
-        });
-      });
-
-      describe('is "user"', function() {
```
Comment (0xF013) [inline frontend/tests/unit/tests/work_packages/controllers/details-tab-overview-controller-test.js:504]: ok, I think this should stay

**Comment by myabc** [comment]
@0xF013 since #2427 was merged, you'll need to rebase.

**Comment by 0xF013** [comment]
@myabc done

**Comment by 0xF013** [comment]
@ulferts I have removed some of your code (version related) again

**Comment by ulferts** [comment]
@0xF013 I fear that removing the version specific custom field code removed the functionality to display the version as desired. So when you have a custom field defined on work packages instead of the id being displayed like it is with your branch:

image

It should be displayed like this:

image

(Please ignore the bug on the user custom field which should have been fixed in current dev.

Other than that (and the bug concerning setting the version to null which I will take a look at) it is looking almost good but there is one problem: The link placed behind the version should not point to the API href (/api/v3/versions/:id) but to the one in the front end (/versions:id - without the /api/v3 prefix).

**Comment by ulferts** [comment]
Additionally, when merging the bugfix for null links (#2430), then after setting the version to null, the label of the former version remains without it being a link. It should be set to "-" which it is after reloading.

image

**Comment by 0xF013** [comment]
@ulferts I reverted most of your changes and I have also rebased on #2430

**Comment by ulferts** [comment]
Thanks @0xF013

**Comment by 0xF013** [comment]
@ulferts as @myabc noted, https://github.com/opf/openproject/pull/2429/files#diff-104a968c3361c7d252941ec3853ae89aL63 this needs to be cleared of useless comments. Should I issue a separate PR for that?

**Comment by ulferts** [comment]
@0xF013, yes please do.

**Pull request merged**